arches/app/models/concept.py

```diff
@@ -1264,8 +1265,18 @@ def _findNarrower(val, path, rec):
                             _findNarrower(child, path, rec)
                 val.children.sort(key=lambda x: (x.sortorder, x.text))
 
-        for row in rows:
-            rec = dict(list(zip(column_names, row)))
+        def best_language_last(rec):

```

Comment:
We now have several places in the code that try to score/rank the match of a concept label for a given language.  

- https://github.com/archesproject/arches/blob/master/arches/app/models/concept.py#L797
- https://github.com/archesproject/arches/blob/master/arches/app/models/concept.py#L1403
- https://github.com/archesproject/arches/blob/master/arches/app/utils/i18n.py#L52

It would be nice to centralize this logic into a shared method if possible.

Reply:
Smart. After our huddle I added a `rank_label()` util in [72e3ccd](https://github.com/archesproject/arches/pull/9849/commits/72e3ccd8ad83f8dfd08d5ad1d9103386ceb81a7d) and was able to use it in the first two links you posted. (The third felt like it was okay as is.)